File: 0392-Is Subsequence/c++/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int subIndex = 0;
        char targetPtr = s[subIndex];
        string target = "";
        
        for(char iter: t) {
            if(iter == targetPtr) {
                target += iter;
                targetPtr = s[++subIndex];
            }
        }
        
        if(target == s) 
            return true;
        return false;
    }
};

// Runtime: 0 ms, faster than 100.00% of C++ online submissions for Is Subsequence.
// Memory Usage: 6.3 MB, less than 86.44% of C++ online submissions for Is Subsequence.

File: 0392-Is Subsequence/golang/solution.go
package golang

func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}
	subIndex := 0
	targetPtr := s[subIndex]

	for _, char := range t {
		if char == rune(targetPtr) {
			subIndex++
			if subIndex == len(s) {
				return true
			}
			targetPtr = s[subIndex]

		}
	}

	return false
}

// Runtime: 0 ms, faster than 100.00% of Go online submissions for Is Subsequence.
// Memory Usage: 2 MB, less than 39.31% of Go online submissions for Is Subsequence.
